src/screens/Customers/AddCustomer.tsx

```diff
@@ -1,38 +1,116 @@
-import React from 'react';
-import { Typography } from '@material-ui/core';
+import { createStyles, FormControl, InputLabel, MenuItem, Select, Theme } from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps } from 'react-router-dom';
 import BaseScreen from '../../components/BaseComponents/BaseScreen';
-import TextField from '../../components/TextField';
-import Checkbox from '../../components/Checkbox';
 import Button from '../../components/Button';
-import { withStyles, createStyles, Theme } from '@material-ui/core/styles';
-import { RouteComponentProps } from 'react-router-dom';
+import Checkbox from '../../components/Checkbox';
+import TextField from '../../components/TextField';
+import { CustomerRecord, SiteRecord } from '../../lib/airtable/interface';
+import { createCustomer } from '../../lib/airtable/request';
+import { addCustomer } from '../../lib/redux/siteData';
+import { RootState } from '../../lib/redux/store';
+
+// TODO: Move to utils file?
+const EMPTY_CUSTOMER : CustomerRecord= {
+    name: '',
+    meterNumber: 0,
+    tariffPlansId: '',
+    isactive: false,
+    hasmeter: false,
+    outstandingBalance: '',
+    meterReadingIds: [],
+    meterReadings: [],
+    paymentIds: [],
+    payments: [],
+    customerUpdateIds: [],
+    customerUpdates: [],
+    totalAmountBilledfromInvoices: 0,
+    totalAmountPaidfromPayments: 0,
+}
 
 const styles = (theme: Theme) =>
   createStyles({
     content: {
       color: theme.palette.text.primary,
     },
+    formControl: {
+      width: '100%',
+    },
   });
 
 interface AddCustomerProps extends RouteComponentProps {
-  classes: { content: string };
+  classes: { content: string, formControl: string};
   location: any;
+  currentSite: SiteRecord;
 }
 
+
 function AddCustomer(props: AddCustomerProps) {
-  const { classes } = props;
+  const { classes, currentSite } = props;
+
+  const [selectedTariffPlan, setSelectedTariffPlan] = useState("");
+  const [customerName, setCustomerName] = useState("");
+  const [meter, setMeter] = useState("");
+  const [meterChecked, setMeterChecked] = useState(false);
+  const [customerInactive, setCustomerInactive] = useState(false);
+
+  const handleSelectTariffPlan = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setSelectedTariffPlan(event.target.value as string);
+  }
+
+  const handleSubmit = (event: React.MouseEvent) => {
+    // Prevent page refresh on submit
+    event.preventDefault();
+
+    const customer = EMPTY_CUSTOMER;
+    customer.name = customerName;
+    customer.meterNumber = parseInt(meter);
+    customer.isactive = !customerInactive;
+    customer.tariffPlansId = selectedTariffPlan;
+    addCustomer(customer);
+    
+    const cust = {
+      ...customer,
+      sitesId: currentSite.rid
```

Comment:
NIT: comment an explanation on why this is needed. It might also be cleaner to just not store it as a variable (since then you'd have to think of a name) and instead just call the `createCustomer` function and do the object instantiation on the same line.

```
createCustomer({
...customer,
sitesId: currentSite.rid
});
```